app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/provision/cloning.rb

```diff
@@ -29,10 +29,16 @@ def customize_destination
     signal :post_create_destination
   end
 
-  def find_destination_in_vmdb(_ems_ref)
+  def find_destination_in_vmdb(ems_ref)
     return if phase_context[:cloud_api_completion_time].nil? || source.ext_management_system.last_refresh_date < phase_context[:cloud_api_completion_time]
 
-    source.ext_management_system&.vms_and_templates&.find_by(:ems_id => options[:src_ems_id].try(:first), :name => options[:vm_name], :template => (request_type == 'clone_to_template'))
+    if request_type == 'clone_to_template'
```

Comment:
Since this PR is focused on fixing a bug I continued with this `if request_type` pattern, but would like to come back and split this module into two*, one for each request_type:
- `template` (VM Provision)
- `clone_to_template` (VM Publish)

*note: there's even a third type we haven't implemented yet - `clone_to_vm` (VM Clone) - which would make using a single module with conditionals even worse.